set.mm

```diff
@@ -662194,6 +662391,123 @@ number axioms (add ~ ax-10 , ~ ax-11 , ~ ax-13 , ~ ax-nul , and remove
+      UISUJUMUTURVBUTURKAUTVCFHIUTVDFHIZVFFHIAFHIBCAVCGFHTVCVDFHTVCVFFHTVCAFHTU
+      TRUTVGKZVLKZVDGVGVJUTVLVKNVNOVMVLPUTVGVLQUNSUOUMUPUQ $.
+    $( $j usage 'sn-nnne0' avoids 'ax-mulcom'; $)
+  $}
+
+  ${
+    $d A x y $.  $d N x y $.  $d ph x y $.
+    renegmulnnass.a $e |- ( ph -> A e. RR ) $.
+    renegmulnnass.n $e |- ( ph -> N e. NN ) $.
+    $( Move multiplication by a natural number inside and outside negation.
+       (Contributed by SN, 25-Jan-2025.) $)
+    renegmulnnass $p |- ( ph -> ( ( 0 -R A ) x. N ) = ( 0 -R ( A x. N ) ) ) $=
+      ( vx wcel cc0 cresub co cmul wceq caddc oveq2 oveq2d eqeq12d syl ad2antrr
+      c1 cr vy cn cv weq rernegcl ax-1rid eqtr4d wa 0red nnre ad2antlr remulcld
+      simpr readdsub syl3anc readdid2 oveq1d eqtr3d resubsub4 3eqtr4d nnadd1com
+      3eqtrd cc recnd 1cnd nncn adddid eqtrd nnindd mpdan ) ACUBGHBIJZCKJZHBCKJ
+      ZIJZLZEAVKFUCZKJZHBVPKJZIJZLVKSKJZHBSKJZIJZLVKUAUCZKJZHBWCKJZIJZLZVKWCSMJ
+      ZKJZHBWHKJZIJZLVOFUACVPSLZVQVTVSWBVPSVKKNWLVRWAHIVPSBKNOPFUAUDZVQWDVSWFVP
+      WCVKKNWMVRWEHIVPWCBKNOPVPWHLZVQWIVSWKVPWHVKKNWNVRWJHIVPWHBKNOPVPCLZVQVLVS
+      VNVPCVKKNWOVRVMHIVPCBKNOPAVTVKWBAVKTGZVTVKLABTGZWPDBUEQZVKUFQZAWABHIAWQWA
+      BLDBUFQZOUGAWCUBGZUHZWGUHZVTWDMJZHWEWAMJZIJZWIWKXCVKWDMJZHWEBMJZIJZXDXFXC
+      XGVKWFMJZWFBIJZXIXCWDWFVKMXBWGUMOXCHWFMJZBIJZXJXKXCHTGZWFTGZWQXMXJLXCUIZX
+      CWETGZXOXCBWCAWQXAWGDRZXAWCTGAWGWCUJUKULZWEUEQZXRHWFBUNUOXCXLWFBIXCXOXLWF
+      LXTWFUPQUQURXCXNXQWQXKXILXPXSXRHWEBUSUOVBAXDXGLXAWGAVTVKWDMWSUQRAXFXILXAW
+      GAXEXHHIAWABWEMWTOORUTXCWIVKSWCMJZKJZXDXAWIYBLAWGXAWHYAVKKWCVAOUKXCVKSWCA
+      VKVCGXAWGAVKWRVDRXCVEZXAWCVCGAWGWCVFUKZVGVHXCWJXEHIXCBWCSABVCGXAWGABDVDRY
+      DYCVGOUTVIVJ $.
+  $}
+
+  $( ~ elznn0nn restated using ~ df-resub .  (Contributed by SN,
+     25-Jan-2025.) $)
+  reelznn0nn $p |-
+    ( N e. ZZ <-> ( N e. NN0 \/ ( N e. RR /\ ( 0 -R N ) e. NN ) ) ) $=
+    ( cz wcel cn0 cr cneg cn wa cc0 cresub elznn0nn cmin df-neg wceq resubeqsub
+    wo co 0re mpan eqtr4id eleq1d pm5.32i orbi2i bitri ) ABCADCZAECZAFZGCZHZPUE
+    UFIAJQZGCZHZPAKUIULUEUFUHUKUFUGUJGUFUGIALQZUJAMIECUFUJUMNRIAOSTUAUBUCUD $.
+
+  $( Multiplication is commutative for nonnegative integers.  Proven without
+     ~ ax-mulcom .  (Contributed by SN, 25-Jan-2025.) $)
+  nn0mulcom $p |- ( ( A e. NN0 /\ B e. NN0 ) -> ( A x. B ) = ( B x. A ) ) $=
+    ( cn0 wcel cn cc0 wceq wo cmul co elnn0 cr remul02 remul01 eqtr4d syl oveq1
+    oveq2 eqeq12d syl5ibrcom nnmulcom impcom jaoian sylanb nn0re jaodan sylan2b
+    nnre imp ) BCDACDZBEDZBFGZHABIJZBAIJZGZBKUJUKUOULUJAEDZAFGZHUKUOAKUPUKUOUQA
+    BUAUKUQUOUKUOUQFBIJZBFIJZGZUKBLDZUTBUHVAURFUSBMBNOPUQUMURUNUSAFBIQAFBIRSTUB
+    UCUDUJULUOUJUOULAFIJZFAIJZGZUJALDZVDAUEVEVBFVCANAMOPULUMVBUNVCBFAIRBFAIQSTU
+    IUFUG $.
+  $( $j usage 'nn0mulcom' avoids 'ax-mulcom'; $)
+
+  $( Lemma for ~ zmulcom .  (Contributed by SN, 25-Jan-2025.) $)
+  zmulcomlem $p |- ( ( ( A e. RR /\ ( 0 -R A ) e. NN ) /\ B e. NN ) ->
```

Comment:
I know it's a lemma, but it seems unnatural to exclude zero. Is the proof longer with `NN0` in place of `NN` ? If yes, then it's probably a symptom of some missing library lemmas.